base/server/healthcheck/pki/server/healthcheck/clones/plugin.py

```diff
@@ -0,0 +1,310 @@
+        output = None
+        try:
+            output = subprocess.check_output(command, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            output = e.output.decode('utf-8')
+            return output
+
+        output = output.decode('utf-8')
+
+        return output
+
+    def get_security_domain_data(self, host, port):
+        domain_data = None
+
+        try:
+            connection = PKIConnection(protocol='http',
+                                       hostname=host,
+                                       port=port,
+                                       verify=False)
+
+            securityDomainClient = SecurityDomainClient(connection)
+            domain_data = securityDomainClient.get_domain_info()
+
+        except BaseException as e:
+            logger.error("Internal server error %s", e)
+            return domain_data
+
+        systems = domain_data.subsystems
+        for s in systems.values():
+            for h in s.hosts.values():
+                if s.id == 'CA':
+                    if h.Clone == 'TRUE':
+                        self.clone_cas.append(h)
+                    else:
+                        self.master_cas.append(h)
+                    continue
+
+                if s.id == 'KRA':
+                    if h.Clone == 'TRUE':
+                        self.clone_kras.append(h)
+                    else:
+                        self.master_kras.append(h)
+                    continue
+
+                if s.id == 'OCSP':
+                    if h.Clone == 'TRUE':
+                        self.clone_ocsps.append(h)
+                    else:
+                        self.master_ocsps.append(h)
+                    continue
+
+                if s.id == 'TPS':
+                    if h.Clone == 'TRUE':
+                        self.clone_tpss.append(h)
+                    else:
+                        self.master_tpss.append(h)
+                    continue
+
+                if s.id == 'TKS':
+                    if h.Clone == 'TRUE':
+                        self.clone_tkss.append(h)
+                    else:
+                        self.master_tkss.append(h)
+                    continue
+
+        return domain_data
+
+    def get_security_domain_ca(self):
+        sec_domain = None
+        sechost = None
+        secport = None
+        for subsystem in self.instance.get_subsystems():
+            if(subsystem.type == 'CA'):
+                # make sure this CA is the security domain
+                service_host = subsystem.config.get('service.machineName')
+                service_port = subsystem.config.get('service.unsecurePort')
+                sechost = subsystem.config.get('securitydomain.host')
+                secport = subsystem.config.get('securitydomain.httpport')
+
+                if sechost == service_host and secport == service_port:
+                    sec_domain = subsystem
+                    break
```

Comment:
Do we need to loop for all subsystems? Can we do this?

````
ca_subsystem = instance.get_subsystem('ca')
if ca_subsystem:
    ...
````

Reply:
Fixes as requested.